applications/nrf_desktop/src/modules/usb_state.c

```diff
@@ -28,22 +28,41 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
 #include "config_event.h"
 #include "power_event.h"
 
+#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
+  #include "usb_state_def.h"
+#else
+  #define USB_SUBSCRIBER_BITMASK(i, unused)	UINT16_MAX,
+
+  /* By default every USB HID instance subscribes for all the reports. */
+  static const uint16_t usb_hid_selected_report_bitmask[] = {
```

Comment:
as mentioned earlier - this variable has no purpose other then be used in asserts
please remove it and move the code (if required) to the initialization function

```
#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
  #include "usb_state_def.h"
#endif

(...)
static void verify_report_bm(void)
{
#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
   if (IS_ENABLED(DEBUG)) {
      testingstuff
   }
#endif
}

static uint32_t get_report_bm(size_t hid_id)
{
   // put any build asserts here to logically separate the code
#if CONFIG_DESKTOP_USB_SELECTIVE_REPORT_SUBSCRIPTION
    return table_entry[i];
#else
    return UINT32_MAX;
#endif
}

(...)

usb_hid_device[i].report_bm = get_report_bm(i);
```

Reply:
ok, done